src/libs/ReportActionsUtils.js

```diff
@@ -403,11 +403,14 @@ function getLastVisibleAction(reportID, actionsToMerge = {}) {
         ...allReportActions[reportID],
         ...updatedActionsToMerge,
     });
-    const visibleActions = actions.filter((action) => shouldReportActionBeVisibleAsLastAction(action));
+
+    const parentReportAction = getParentReportAction(allReports[reportID]);
+    const visibleActions = actions.filter((action) => shouldReportActionBeVisibleAsLastAction(action) && (!isDeletedAction(parentReportAction) || !isModifiedExpenseAction(action)));
```

Comment:
We already have the check for `!isDeletedAction` here

https://github.com/Expensify/App/blob/12d366931012b0092b924b688924726930398ac4/src/libs/ReportActionsUtils.js#L383-L386

So I guess the check won't be reaching `!isModifiedExpenseAction(action)`, no?